oletools/util/log_helper/log_helper.py

```diff
@@ -0,0 +1,194 @@
+            self._set_logger_level(logger, logging.NOTSET)
+
+        # add a JSON formatter to the root logger, which will be used by every logger
+        if self._use_json:
+            _root_logger_wrapper.set_formatter(JsonFormatter())
+            print('[')
+
+    def end_logging(self):
+        """
+        Must be called at the end of the main function if the caller wants
+        json-compatible output
+        """
+        if not self._is_enabled:
+            return
+        self._is_enabled = False
+
+        # end logging
+        self._all_names = set()
+        logging.shutdown()
+
+        # end json list
+        if self._use_json:
+            print(']')
+        self._use_json = False
+
+    def _get_except_hook(self, old_hook):
+        """
+        Global hook for exceptions so we can always end logging.
+        We wrap any hook currently set to avoid overwriting global hooks set by oletools.
+        Note that this is only called by enable_logging, which in turn is called by
+        the main() function in oletools' scripts. When scripts are being imported this
+        code won't execute and won't affect global hooks.
+        """
+        def hook(exctype, value, traceback):
+            self.end_logging()
+            old_hook(exctype, value, traceback)
+
+        return hook
+
+    def _get_or_create_logger(self, name, level, handler=None):
+        """
+        Get or create a new logger. This newly created logger will have the
+        handler and level that was passed, but if it already exists it's not changed.
+        We also wrap the logger in an adapter so we can easily extend its functionality.
+        """
+
+        # logging.getLogger creates a logger if it doesn't exist,
+        # so we need to check before calling it
+        if handler and not self._log_exists(name):
+            logger = logging.getLogger(name)
+            logger.addHandler(handler)
+            self._set_logger_level(logger, level)
+        else:
+            logger = logging.getLogger(name)
+
+        # Keep track of every logger we created so we can easily change
+        # their levels whenever needed
+        self._all_names.add(name)
+
+        adapted_logger = OletoolsLoggerAdapter(logger, None)
+        adapted_logger.set_json_enabled_function(lambda: self._use_json)
+
+        return adapted_logger
+
+    @staticmethod
+    def _set_logger_level(logger, level):
+        """
+        If the logging is already initialized, we set the level of our logger
+        to 0, meaning that it will reuse the level of the root logger.
+        That means that if the root logger level changes, we will keep using
+        its level and not logging unnecessarily.
+        """
+
+        # if this log was wrapped, unwrap it to set the level
+        if isinstance(logger, OletoolsLoggerAdapter):
+            logger = logger.logger
+
+        if _root_logger_wrapper.is_logging_initialized():
+            logger.setLevel(logging.NOTSET)
```

Comment:
This is an edge case: suppose we create a logger and call enable_logging(). The logger will have its level set to `logging.NOTSET`, meaning it will fallback to the root logger's level when logging. Now suppose we create another logger after that - this function will be called and try to set the level to `logging.CRITICAL+1`, but we probably don't want that since we already enabled logging.